lib/oauth2-service.js

```diff
@@ -137,10 +137,17 @@ class OAuth2Service extends EventEmitter {
 
     let xfn;
     let { scope } = req.body;
+    const credentials = basicAuth(req);
+    const clientId = credentials ? credentials.name : null;
 
     switch (req.body.grant_type) {
       case 'client_credentials':
-        xfn = scope;
+        xfn = (header, payload) => {
+          Object.assign(payload, {
+            aud: clientId,
+            scope,
+          });
+        };
```

Comment:
`aud` claim is the `client_id` only for the `id_token`. This makes sense, as the `aud` purpose is to state: "for whom the token was issued".
Normally, the ID token is intended for the client app, so `client_id` is fine.
For an `access_token`, like in this case, the audience is normally the `resource_server`. Which can be your backend api which will read the access_token and use it to allow or reject access.

In any case, if you still want to do it, your test code has full control on the claims issued, check [this test](https://github.com/axa-group/oauth2-mock-server/blob/0abed386e242b1cc5c8b598fe0ddee438d3f6e5a/test/oauth2-issuer.test.js#L101-L115)

Reply:
@acasella I agree, this is something use only on the company I worked for and I think it does not make sense to add it here, please close the PR, I will keep this just in my fork.